### ISSUE TITLE
Runaway promise error when using this library with Bluebird

### ISSUE THREAD

**Issue by genui-eeshan** [FormidableLabs/rapscallion#119]
When using this library with Bluebird, Bluebird throws a runaway promise error for the following: 

```
(node:1528) Warning: a promise was created in a handler at node_modules/rapscallion/lib/consumers/node-stream.js:42:9 but was not returned from it, see http://goo.gl/rRqMUw
    at Function.Promise.cast (node_modules/bluebird/js/release/promise.js:196:13)
```

More information about the error can be found here: http://goo.gl/rRqMUw

**Comment by genui-eeshan** [comment]
A simple fix that worked locally was adding ```return null``` before the catch on line 44.

**Comment by ryan-roemer** [comment]
We'd ❤️  a PR with a regression test!

**Comment by genui-eeshan** [comment]
@ryan-roemer Someone created a PR for this. #111 

What's it missing to be pushed in?

**Comment by ryan-roemer** [comment]
@genui-eeshan -- Glad to hear it's the same issue since L44 isn't in that PR. Added a reference in https://github.com/FormidableLabs/rapscallion/pull/111 to close this issue when merged.

**Comment by ryan-roemer** [comment]
Fixed in #124 if I'm tracking things correctly.

@ethersage -- is that right?

**Comment by ethersage** [comment]
@ryan-roemer Correct. Have not released a patch version yet though.
